custom_components/hyxi_cloud/coordinator.py

```
@@ -147,6 +176,25 @@ async def _async_update_data(self):
             await self._async_sync_device_metadata(devices)
             return devices
 
+        except (ClientError, TimeoutError, UpdateFailed) as err:
+            try:
+                cached_devices = await self.device_store.async_load()
+                if cached_devices:
+                    self.hyxi_metadata["last_error"] = str(err)
+                    self.hyxi_metadata["api_status"] = "Offline"
+                    _LOGGER.exception(
+                        "HYXI Cloud API fetch failed: %s. Falling back to %d cached devices from storage.",
+                        err,
```

Ruff (0.15.20)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)